text/0000-fix-fees.md

```diff
@@ -0,0 +1,100 @@
+- Title: fix-fees
+- Authors: [John Tromp](mailto:john.tromp@gmail.com)
+- Start date: August 23, 2020
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000) 
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+# Summary
+[summary]: #summary
+
+Change Grin's minimum relay fees to be weight proportional and make output creation cost about a Grin-cent.
+
+# Motivation
+[motivation]: #motivation
+
+The current fee requirements suffer from being somewhat arbitrary, and not miner incentive compatible.
+They are not even linear; the fees required for the aggregate of two transactions is not necessarily equal to the sum
+of required fees.
+The current (and foreseeable) low price of Grin makes spamming the UTXO set rather cheaper than desired.
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+Grin constrains the size of blocks by a maximum block weight, which is a linear combination of the number of inputs, outputs, and kernels.
+When blocks fill up, miners are incentivized to pick transactions in decreasing order of fees per weight.
+Ideally then, a transaction that is more preferred by miners, is also more likely to be relayed.
+If a transaction A doesn't satisfy the minimum relay fees, while another transaction B does, but miners would rather include A,
+then A's publisher will have to transmit it directly to miners, which is undesirable.
+We therefore let the minimum relay fee be proportional to the weight of the transaction, minimizing arbitrariness.
+We can calibrate the fee system by stipulating that creating one output should cost at least one Grin-cent (formerly 0.4 Grin-cent).
+Linearity is another desirable property, which for instance allows the two parties in a payjoin to each pay their own input and output fees,
+while splitting the kernel fee.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The minimum relay fee of a transaction shall be proportional to Transaction::weight\_as\_block,
+which uses weights of BLOCK\_INPUT\_WEIGHT = 1, BLOCK\_OUTPUT\_WEIGHT = 21, and BLOCK\_KERNEL\_WEIGHT = 3,
+which correspond to the nearest multiple of 32 bytes that it takes to serialize.
+Formerly, we used Transaction::weight\_as\_block,
```

Comment:
did you mean something else than `Transaction::weight\_as\_block` here? This one has weights 1, 21, 3 as mentioned above.